sklearn/preprocessing/_data.py

```diff
@@ -779,7 +803,38 @@ def partial_fit(self, X, y=None):
             if self.with_std:
                 # First pass
                 if not hasattr(self, 'scale_'):
-                    self.mean_, self.var_ = mean_variance_axis(X, axis=0)
+                    if sample_weight is not None:
+                        from sklearn.utils.extmath import safe_sparse_dot
```

Comment:
why don't you use _incremental_weighted_mean_and_var here too?

Reply:
`_incremental_weighted_mean_and_var` does not work for sparse data. But I agree we should probably either extend that helper function to add support for sparse data or write a new help function `_sparse_incremental_weighted_mean_and_var` and unittest them together.